### PR TITLE
Exclude checking the pod statuses with the name prefix "report-status-to-provider-"

### PR DESCRIPTION
**Pull request by yitzhak12**
In the PR, I implemented the following:
- In the functions `check_pods_after_node_replacement` and `check_pods_in_running_state`, exclude checking the pod statuses with the name prefix `report-status-to-provider-`.
- Add the param `exclude_pod_name_prefixes` to the functions `check_pods_in_statuses` and `get_pods_in_statuses` for excluding the pod names with specific prefixes.

### CODE REVIEW

**Comment by yitzhak12** [comment]
I needed to increase the timeout of waiting for the machine to reach the Failed state, as sometimes it takes almost 10 minutes.

**Comment by yitzhak12** [comment]
The pr validation job also passed when running the test on an MS consumer cluster: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/1932/.

**Review by ebondare — APPROVED** [review]

**Review by suchita-g — APPROVED** [review]

File: ocs_ci/ocs/resources/pod.py [hunk 1]
```diff
@@ -2608,14 +2609,17 @@ def get_osd_pod_id(osd_pod):
...
+    namespace=config.ENV_DATA["cluster_namespace"],
+    exclude_pod_name_prefixes=None,
 ):
     """
     Get all the pods in specific statuses
 
     Args:
         status_options (list): The list of the status options.
         namespace (str): Name of cluster namespace(default: config.ENV_DATA["cluster_namespace"])
+        exclude_pod_name_prefixes (list): The list of the pod name prefixes to exclude from the pods to get
```
Comment (dahorak) [inline ocs_ci/ocs/resources/pod.py:2622]: This line and few other lines in the docs strings are longer than 120 characters (in the [Coding Guidelines](https://ocs-ci.readthedocs.io/en/latest/coding_guidelines.html) we have even 79 characters, but [in tox.ini](https://github.com/red-hat-storage/ocs-ci/blob/master/tox.ini#L42) there is 120). I'm little bit confused how it passed the tox though.

**Review by dahorak — CHANGES_REQUESTED** [review]
~~Some of the docstrings have too long lines.~~
I've wrongly check the lines length.

**Review by dahorak — APPROVED** [review]

**Pull request merged**